golang/outbound.go

```diff
@@ -56,12 +56,17 @@ func (c *Connection) beginCall(ctx context.Context, serviceName string) (*Outbou
 		return nil, err
 	}
 
+	headers := callHeaders{
+		CallerName: c.localPeerInfo.ProcessName,
```

Comment:
This should be the name of the service making the call, once you've added the Hyperbahn registration.  It's used for circuit breaking and backrouting, so I don't think we ever want it to be the process name (which is an ephemeral unroutable entity).